presto-main/src/main/java/com/facebook/presto/metadata/MetadataManager.java

```diff
@@ -528,6 +529,24 @@ public void dropTable(Session session, TableHandle tableHandle)
     }
 
     @Override
+    public void beginQuery(Session session, Collection<TableHandle> tableHandles)
+    {
+        Set<String> distrinctConnectors = retrieveConnectorIds(tableHandles);
+
+        for (String connectorId : distrinctConnectors) {
+            ConnectorEntry entry = connectorsById.get(connectorId);
+            ConnectorSession connectorSession = session.toConnectorSession(entry.getCatalog());
+            ConnectorMetadata metadata = entry.getMetadata(session);
+            metadata.beginQuery(connectorSession);
```

Comment:
This was discussed with @electrum and he proposed just `beginQuery(Session)` method, have a look here:
https://github.com/prestodb/presto/pull/5122#issuecomment-229749167
My previous version (see PR above) had `TableHandle` circulating back and forth between notification and connector but it was not generic enough.
Moreover, in some calls, namely some DDLs, it may not have sense to speak of `TableHandle` in such a notification.